code/game/machinery/portable_turret.dm

```diff
@@ -575,7 +586,7 @@ var/list/turret_icons
 		if(TURRET_SECONDARY_TARGET)
 			secondarytargets += L
 
-/obj/machinery/porta_turret/proc/assess_living(var/mob/living/L)
+/obj/machinery/porta_turret/proc/assess_living(var/mob/living/L) //TODO: optimize
```

Comment:
Can't select following code for suggestion.
```
/obj/machinery/porta_turret/proc/assess_living(mob/living/L)
	if(istype(L) && check_trajectory(L, src) && L.invisibility >= INVISIBILITY_LEVEL_ONE)
		if(isblitzshell(L))
			return TURRET_PRIORITY_TARGET
		if(check_anomalies && (isanimal(L) || issmall(L)))
			return TURRET_SECONDARY_TARGET
		if(emagged || hackfail || (check_synth && assess_perp(L) > 3 && !issilicon(L)))
			if(L.lying)
				return lethal ? TURRET_SECONDARY_TARGET : TURRET_NOT_TARGET
			else
				return lethal ? TURRET_PRIORITY_TARGET : TURRET_SECONDARY_TARGET
		if(ishuman(L))
			var/obj/item/card/id/id_card = L.GetIdCard()
			if(!L.stat && !iscuffed(L) && (!id_card || !(id_card.registered_name in registered_names)))
				if(L.lying)
					return lethal ? TURRET_SECONDARY_TARGET : TURRET_NOT_TARGET
				else
					return lethal ? TURRET_PRIORITY_TARGET : TURRET_SECONDARY_TARGET

	return TURRET_NOT_TARGET
```

Reply:
im not sure we want to sacrifice our colony ally targetting for this

i might use this later but modify it to work better with our code